xrviz/dashboard.py

```diff
@@ -81,8 +98,25 @@ def create_plot(self, *args):
             if self.var in list(sel_data.coords):  # When a var(coord) is plotted wrt itself
                 sel_data = sel_data.to_dataset(name=f'{sel_data.name}_')
 
+            c_low_val = self.kwargs['cmap lower limit']
+            c_upp_val = self.kwargs['cmap upper limit']
+
+            # It is better to set initial values as 0.1,0.9 rather than 0,1 i.e. (min, max)
+            # to get a color balanced graph
+            c_lim_lower = float(c_low_val) if c_low_val else sel_data.quantile(0.1)
+            c_lim_upper = float(c_upp_val) if c_upp_val else sel_data.quantile(0.9)
+
+            color_range = {sel_data.name: (c_lim_lower, c_lim_upper)}
+
+            if not c_low_val:  # if initial values are empty
+                self.control.style.lower_limit.value = str(c_lim_lower.values.round(5))
+                self.control.style.upper_limit.value = str(c_lim_upper.values.round(5))
+
+            if color_scale is not 'linear':
+                sel_data = getattr(numpy, color_scale)(sel_data)  # Color Scaling
```

Comment:
So is this in the original or scaled values?

Reply:
Color_scaling in done on the original data. `Cmap Limits` are calculated using original and not scaled data.

Reply:
This has been changed now.

Reply:
I didn't follow your reasoning here, perhaps this is a good point for discussion.

Reply:
Initially on color_scaling, quantiles(0.1,0.9) of actual data was filled in `cmap limits`. It resulted in completely dark graph for most of the color scales, since on scaling the data values change. Now the values of `cmap limits` will become empty on selection of new scale. After clicking plot, the values are filled according to quantiles of scaled values due to which graph will have proper colors. (as explained [below](https://github.com/intake/xrviz/pull/13#issuecomment-507574327)).